Lib/string.py

```diff
@@ -83,7 +83,7 @@ class Template(metaclass=_TemplateMetaclass):
     # but without ASCII flag.  We can't add re.ASCII to flags because of
     # backward compatibility.  So we use local -i flag and [a-zA-Z] pattern.
     # See https://bugs.python.org/issue31672
-    idpattern = r'(?-i:[_a-zA-Z][_a-zA-Z0-9]*)'
+    idpattern = r'(?a:[_a-zA-Z][_a-zA-Z0-9]*)'
```

Comment:
How about changing `a-zA-Z` to `a-z`?

Reply:
And update the comment above.

Reply:
Actually, I don't think we can remove `A-Z` from the pattern because that *could* be seen as a backward compatibility break.  E.g. what if there's code out there that sets `flags=0`?  Then the regexp wouldn't be case-insensitive and the removal of `A-Z` would break that code.

I think it *is* safe to swap `a` for `-i` though, so I'll do that.  I'll also update the comment and, as I just also noticed, the documentation.

Reply:
This will just return the situation before the initial fix of this issue, but with correctly fixed bug. Initially the pattern was `r'[_a-z][_a-z0-9]*'`, later it was changed to `r'(?-i:[_a-zA-Z][_a-zA-Z0-9]*)'`, and now we can use `r'(?a:[_a-z][_a-z0-9]*)'` which is a fixed (ASCII-only) version of `r'[_a-z][_a-z0-9]*'`.